fastai2/distributed.py

```diff
@@ -114,15 +134,15 @@ def after_fit(self):
 @patch
 def to_distributed(self: Learner, cuda_id):
     self.add_cb(DistributedTrainer(cuda_id))
-    if rank_distrib() > 0: self.remove_cb(self.progress)
+    if rank_distrib() > 0: self.remove_cb(ProgressCallback)
     return self
 
 # Cell
 @patch
 def detach_distributed(self: Learner):
     if num_distrib() <=1: return self
-    if hasattr(self, 'distributed_trainer'): self.remove_cb(self.distributed_trainer)
-    if rank_distrib() > 0 and not hasattr(self, 'progress'): self.add_cb(ProgressCallback())
+    self.remove_cb(DistributedTrainer)
+    if rank_distrib() > 0: self.add_cb(ProgressCallback())
```

Comment:
```suggestion
    if rank_distrib() > 0 and not hasattr(self, 'progress'): self.add_cb(ProgressCallback())
```

Reply:
I think you still need the test here to avoid having the progress callback twice. (If you agree, the change is also needed in the notebook).

Reply:
Hi Sylvain, thanks for reviewing the change.

I thought about keep using the `hasattr()` check, or let `add_cb()`'s assertion to complain duplicate registration loud and clear, and reveal  the problem that "was the callback properly removed before?" 

```
def add_cb(self, cb):
        ...
        assert not old or isinstance(old, type(cb)), f"self.{cb.name} already registered"
```
What do u think?  I can put the `hasattr()` back if that's deemed more useful.

Reply:
The user won't be able to do anything about it since they are not adding the callback themselves. It's cleaner if the function does not add it if it's already there.

Reply:
Got it, thx!

Reply:
> the change is also needed in the notebook

Done.